trie/trie.go

```diff
@@ -574,18 +669,28 @@ func (t *Trie) resolve(n node, prefix []byte) (node, error) {
 	return n, nil
 }
 
-func (t *Trie) resolveHash(n hashNode, prefix []byte, epoch uint16) (node, error) {
+func (t *Trie) resolveHash(n hashNode, prefix []byte) (node, error) {
 	hash := common.BytesToHash(n)
 	if t.db == nil {
 		return nil, fmt.Errorf("empty trie database")
 	}
 	if node := t.db.node(hash); node != nil {
-		node.setEpoch(epoch)
 		return node, nil
 	}
 	return nil, &MissingNodeError{NodeHash: hash, Path: prefix}
 }
 
+func (t *Trie) nodeExpired(n node, prefix []byte) (bool, error) {
```

Comment:
please help modify the `ExpiredNodeError` struct to include `Addr` and `Key`. It may look something like this:

```
type ExpiredNodeError struct {
	ExpiredNode node   // node of the expired node
        Addr        common.Address
        Key         key
	Path        []byte // hex-encoded path to the expired node
	Epoch       uint16
}
```

```
func (t *Trie) nodeExpired(n node, key []byte, prefix []byte) (bool, error) {
	if t.currentEpoch-n.getEpoch() >= 2 {
		return true, &ExpiredNodeError{
			ExpiredNode: n,
                        Addr:        nil
                        Key:         key
			Path:        prefix,
			Epoch:       n.getEpoch(),
		}
	}
	return false, nil
}
```